File: web/projects/portal/src/app/vsobjects/event/filter-table-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { TableCellEvent } from "./table-cell-event";

/**
 * Event used to add a table filter on viewsheet.
 */
export class FilterTableEvent extends TableCellEvent {
   /**
    * The top offset position of the filter relative to the table.
    */
   public top: number;

   /**
    * The left offset position of the filter relative to the table.
    */
   public left: number;

   /**
    * Creates a new instance of <tt>FilterTableEvent</tt>.
    *
    * @param objectName the name of the object.
    * @param index      the column index.
    * @param top        the top position for the filter.
    * @param left       the left position for the filter.
    */
   constructor(objectName: string, index: number, top: number, left: number)
   {
      super(objectName, index);
      this.top = top;
      this.left = left;
   }
}

File: core/src/main/java/inetsoft/web/composer/vs/objects/event/FilterTableEvent.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.composer.vs.objects.event;

/**
 * Class that encapsulates the parameters for filtering a table.
 *
 * @since 12.3
 */
public class FilterTableEvent extends TableCellEvent {
   public int getTop() {
      return top;
   }

   public void setTop(int top) {
      this.top = top;
   }

   public int getLeft() {
      return left;
   }

   public void setLeft(int left) {
      this.left = left;
   }

   @Override
   public String toString() {
      return "FilterTableEvent{" +
         "name='" + this.getName() + '\'' +
         ", top=" + top +
         ", left=" + left +
         ", col=" + getCol() +
         ", row=" + getRow() +
         '}';
   }

   private int top;
   private int left;
}
